### PR TITLE
domain name routing should not be case sensitive

### PR DESCRIPTION
**Pull request by Xianic**

### CODE REVIEW

File: dns_reverse_proxy.go [hunk 1]
```diff
@@ -82,7 +82,7 @@ func main() {
 			}
 			backends = append(backends, backend)
 		}
-		if !strings.HasSuffix(s[0], ".") {
+		if !strings.HasSuffix(strings.ToLower(s[0]), ".") {
```
Comment (StalkR) [inline dns_reverse_proxy.go]: Is it really necessary here? we're checking for "." which does not have uppercase or lowercase.

File: dns_reverse_proxy.go [hunk 2]
```diff
@@ -124,8 +124,10 @@ func route(w dns.ResponseWriter, req *dns.Msg) {
 		dns.HandleFailed(w, req)
 		return
 	}
+
+	lcName := strings.ToLower(req.Question[0].Name)
 	for name, addrs := range routes {
-		if strings.HasSuffix(req.Question[0].Name, name) {
+		if strings.HasSuffix(lcName, name) {
```
Comment (StalkR) [inline dns_reverse_proxy.go:130]: routes could have been configured in uppercase, can you add:
`name := strings.ToLower(name)`

**Review by StalkR — COMMENTED** [review]
Thanks for the change! Indeed, we should be case insensitive since DNS is.

File: dns_reverse_proxy.go (hunk 1 above)
Comment (Xianic) [reply dns_reverse_proxy.go]: Erm, no. This is the wrong place to add it. It should be on line 88 so as to save the routes into the map in lower case. I'll fix this on Tuesday.

**Comment by StalkR** [comment]
Thanks @Xianic !

**Pull request merged**